server/build.gradle

```diff
@@ -298,3 +298,10 @@ tasks.named("licenseHeaders").configure {
     excludes << 'org/apache/lucene/search/RegExp87*'
     excludes << 'org/apache/lucene/search/RegexpQuery87*'
 }
+
+tasks.named("test").configure {
+  /* We need to set this to prevent release builds from failing while the default is different between snapshot and release builds.
+  * In particular, this is needed for IndexNameExpressionResolverTests, which assumes the value is `deprecated`.
+  */
+  systemProperty 'es.system_index_access_behavior', 'deprecated'
```

Comment:
I am not sure about this; shouldn't this be an `assume` in the tests?

Reply:
That sounds like a better solution - I just didn't think of it when I noticed the situation would happen. Will do that instead.

Reply:
I've removed the system property in these gradle files and used `assumeTrue` in the tests in question.